client/src/reducers/controls2.js

```diff
@@ -0,0 +1,333 @@
+// jshint esversion: 6
+
+import _ from "lodash";
+import { World, kvCache } from "../util/stateManager";
+import { parseRGB } from "../util/parseRGB";
+import crossfilter from "../util/typedCrossfilter";
+import * as globals from "../globals";
+
+function createCategoricalAsBooleansMap(world) {
+  const res = {};
+  _.each(world.summary.obs, (value, key) => {
+    if (value.options) {
+      const optionsAsBooleans = {};
+      _.each(value.options, (_value, _key) => {
+        optionsAsBooleans[_key] = true;
+      });
+      res[key] = optionsAsBooleans;
+    }
+  });
+  return res;
+}
+
+const Controls = (
+  state = {
+    // data loading flag
+    loading: false,
+    error: null,
+
+    universe: null,
+
+    // all of the data + selection state
+    world: null,
+    colorName: null,
+    colorRGB: null,
+    categoricalAsBooleansMap: null,
+    obsCrossfilter: null,
+    obsDimensionMap: null,
+
+    colorAccessor: null,
+    colorScale: null,
+
+    opacityForDeselectedCells: 0.2,
+    graphBrushSelection: null,
+    continuousSelection: null,
+    scatterplotXXaccessor: null, // just easier to read
+    scatterplotYYaccessor: null,
+    axesHaveBeenDrawn: false,
+    __storedStateForCelllist1__: null /* will need procedural control of brush ie., brush.extent https://bl.ocks.org/micahstubbs/3cda05ca68cba260cb81 */,
+    __storedStateForCelllist2__: null
+  },
+  action
+) => {
+  /*
+  For now, log anything looking like an error to the console.
+  */
+  if (action.error || /error/i.test(action.type)) {
+    console.error(action.error);
+  }
+
+  switch (action.type) {
+    /*****************************************************
+          Initialization, World/Universe management
+          and data loading.
+    ******************************************************/
+    case "initial data load start": {
+      return { ...state, loading: true };
+    }
+    case "initial data load complete (universe exists)":
+    case "reset World to eq Universe": {
+      /* first light - create world & other data-driven defaults */
+      const { universe } = action;
+      const world = World.createWorldFromEntireUniverse(universe);
+      const colorName = new Array(universe.nObs).fill(globals.defaultCellColor);
+      const colorRGB = _.map(colorName, c => parseRGB(c));
+      const categoricalAsBooleansMap = createCategoricalAsBooleansMap(world);
+      const obsCrossfilter = crossfilter(world.obsAnnotations);
```

Comment:
It seems like we only have one crossfilter, and that will stick. I would prefer the pattern: 

`import Crossfilter from ...`
`const crossfilter = Crossfilter(world.obsAnnotations)`

Uppercase constructor lowercase instance here as well, and that is 'the' crossfilter for the app, and we know to look for that.

Reply:
today we have one, but if we start filtering by var (gene), we would have two.  Hence the naming `obsCrossfilter` - crossfilters on obs annotations.

that said, I'll change it and we can add more clarity when / if we need disambiguation.

Reply:
i assume you want the same change - for the same reason - for obsDimensionMap.  It contains the dimensions on the obs crossfilter.   Will be ambiguous if we ever allow crossfilters on both, but for now it is a moot point.